Hamming distance improvements

AndrzejKurek opened the pull request:
This PR introduces a couple of Hamming distance improvements for return values in Mbed TLS baremetal.

shelib01 commented:
@AndrzejKurek , I think we should also change return code code logic for https://github.com/ARMmbed/mbedtls/pull/3408/files#diff-4acaa6344ae720225c7adfaca894550aR1161
 It return "int" but all the logic and 0/1 numbers simulate true/false

shelib01 commented:
> > @AndrzejKurek , I think we should also change return code code logic for https://github.com/ARMmbed/mbedtls/pull/3408/files#diff-4acaa6344ae720225c7adfaca894550aR1161
> > It return "int" but all the logic and 0/1 numbers simulate true/false
> 
> Could you please elaborate? I reworked the code where `rng_function` is used - to check for exact values instead of just 0/1.

I am talking about API **uECC_generate_random_int**
https://github.com/ARMmbed/mbedtls/blob/8f52a8a8c06ba767ce0be2cf8d1406b7b24878a2/tinycrypt/ecc.c#L1160

It still return 0 or 1 and all paces that use it check the return value as
 if (! uECC_generate_random_int) {
}

AndrzejKurek commented:
> > > @AndrzejKurek , I think we should also change return code code logic for https://github.com/ARMmbed/mbedtls/pull/3408/files#diff-4acaa6344ae720225c7adfaca894550aR1161
> > > It return "int" but all the logic and 0/1 numbers simulate true/false
> > 
> > 
> > Could you please elaborate? I reworked the code where `rng_function` is used - to check for exact values instead of just 0/1.
> 
> I am talking about API **uECC_generate_random_int**
> https://github.com/ARMmbed/mbedtls/blob/8f52a8a8c06ba767ce0be2cf8d1406b7b24878a2/tinycrypt/ecc.c#L1160
> 
> It returns 0 or 1 and all paces that use it check the return value as
> if (! uECC_generate_random_int) {
> }

Updated with requested fixes.

piotr-now commented on include/tinycrypt/ecc.h:
```diff
@@ -155,17 +155,18 @@ extern const uECC_word_t curve_b[NUM_ECC_WORDS];
  * @param num_words IN -- number of words
- * @return a random integer in the range 0 < random < top
+ * @return UECC_SUCCESS in case of success
+ * @return UECC_FAILURE upon failure
```
How did it happen that the earlier API was inconsistent with reality :) ?
